app/views/plantings/show.html.haml

```diff
@@ -67,6 +67,9 @@
     %p= render 'plantings/progress', planting: @planting, show_explanation: true
 
     = render 'plantings/actions', planting: @planting
+    - if !@planting.photos.empty? || (can?(:edit, @planting) && can?(:create, Photo))
```

Comment:
I think this says, you can only have an "add photo" button if there already some photos.
The `if !@planting.photos.empty?`.

Reply:
I think this is harmless, but unnecessary - it says "if the planting has photos OR you can add a photo to the planting...", and then on the next line we check "if you can add a photo to the planting...". So we have the following truth-table:

 - no photos, can't add photo: no button
 - no photos, can add photo: show button
 - has photos, can't add photo: no button
 - has photos, can add photo: show button

So line 70 is doing nothing. Delete line 70, then this is good.